.github/workflows/test.yml

```diff
@@ -6,42 +6,49 @@ on:
       - "master"
   pull_request:
     branches:
-      - "*"
+      - "**"
   workflow_dispatch:
   release:
     types: [published, edited]
 
 jobs:
   tests:
-
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
         python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
 
     steps:
-    - uses: actions/checkout@v4
-    - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v5
-      with:
-        python-version: ${{ matrix.python-version }}
-    - name: Install dependencies
-      run: |
-        python -m pip install --upgrade pip
-        python -m pip install tox tox-gh-actions
-    - name: Cache tox environments
-      id: cache-tox
-      uses: actions/cache@v4
-      with:
-        path: .tox
-        # setup.py and setup.cfg have versioning info that would impact the
-        # tox environment. hashFiles only takes a single file path or pattern
-        # at the moment.
-        key: ${{ runner.os }}-${{ matrix.python-version }}-tox-${{ hashFiles('setup.py') }}-${{ hashFiles('setup.cfg') }} }}
-    - name: Test with tox
-      run: tox
-    - uses: codecov/codecov-action@v4
-      with:
-        token: ${{ secrets.CODECOV_TOKEN }}
-        verbose: true
+      - uses: actions/checkout@v4
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install tox tox-gh-actions
+      - name: Cache tox environments
+        id: cache-tox
+        uses: actions/cache@v4
+        with:
+          path: .tox
+          # setup.py and setup.cfg have versioning info that would impact the
+          # tox environment. hashFiles only takes a single file path or pattern
+          # at the moment.
+          key: ${{ runner.os }}-${{ matrix.python-version }}-tox-${{ hashFiles('setup.py') }}-${{ hashFiles('setup.cfg') }} }}
+      - name: Test with tox
+        run: tox
+      - name: Commit and Push to Pull Request
```

Comment:
@andhus i've decided to add pre-commit trigger into tox 3.8 run, as it's expected for contributors to run tests before commiting. 
and i've added step to commit changes from pre-commit if there are any - if the contributor didn't run it himself
so no separate job is necessary and i can save you some GHA minutes